src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -1,67 +1,211 @@
     Future {
-      //currently, the bucketPath of the clusterRequest are not used - it will be used later as a place to store notebooks and results
+      //currently, the bucketPath of the clusterRequest is not used - it will be used later as a place to store notebooks and results
       val dataproc = new Dataproc.Builder(GoogleNetHttpTransport.newTrustedTransport,
         JacksonFactory.getDefaultInstance, getDataProcServiceAccountCredential)
         .setApplicationName("dataproc").build()
-      val metadata = clusterRequest.labels + ("docker-image" -> dataprocConfig.dataprocDockerImage)
-      val gce = new GceClusterConfig().setMetadata(metadata.asJava).setServiceAccount(clusterRequest.serviceAccount)
-      val initActions = Seq(new NodeInitializationAction().setExecutableFile(dataprocConfig.dataprocInitScriptURI))
-      val clusterConfig = new ClusterConfig().setGceClusterConfig(gce).setInitializationActions(initActions.asJava)
-      val cluster = new Cluster().setClusterName(clusterName).setConfig(clusterConfig)
-      val request = dataproc.projects().regions().clusters().create(googleProject, dataprocConfig.dataprocDefaultZone, cluster)
+
+      val gce = new GceClusterConfig()
+        .setServiceAccount(clusterRequest.serviceAccount)
+        .setTags(List("leonardo"))
+
+      val initActions = Seq(new NodeInitializationAction().setExecutableFile(createBucketUri(bucketName, dc.initActionsScriptName)))
+
+      val worksInstancConfig = new InstanceGroupConfig().setNumInstances(0)
+
+      val clusterConfig = new ClusterConfig()
+        .setGceClusterConfig(gce)
+        .setInitializationActions(initActions.asJava)
+        .setWorkerConfig(worksInstancConfig)
+
+      val cluster = new Cluster()
+        .setClusterName(clusterName)
+        .setConfig(clusterConfig)
+
+      val request = dataproc.projects().regions().clusters().create(googleProject, dc.dataprocDefaultZone, cluster)
+
       try {
         executeGoogleRequest(request)
       } catch {
         case e: GoogleJsonResponseException => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"Google Request Failed: ${e.getDetails.getMessage}"))
       }
     }
   }
+
+  private def updateFirewallRules(googleProject: String) = {
+    val request = new Compute(httpTransport, jsonFactory, getVmServiceAccountCredential).firewalls().get(googleProject, "leonardo-notebooks-rule")
+    try {
+      val response = executeGoogleRequest(request)
+    } catch {
+      case t: GoogleJsonResponseException if t.getStatusCode == 404 => addFirewallRule(googleProject)
+      case e: GoogleJsonResponseException => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"Google Request Failed: ${e.getDetails.getMessage}"))
+    }
+  }
+
+  private def addFirewallRule(googleProject: String) = {    val allowed = new Allowed().set("tcp", "443").setIPProtocol("tcp").setPorts(List("443"))
```

Comment:
missing newline? :)

Reply:
Btw the port is actually configured in `ProxyConfiguration`. Maybe we could read it from there?